File: WindowsFormsApplicationLab1/WindowsFormsApplicationLaba1/WaterTap.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsFormsApplicationLaba1
{
    class WaterTap
    {
        private bool open = false;
        public bool Open { set { open = value; } get { return open; } }

        public void Wash(Apple apple)
        {
            if (Open)
            {
                apple.Dirty = false;
            }
        }
    }
}

File: TPlaba1_Java/src/WaterTap.java

public class WaterTap {
	private boolean open = false;

	public boolean getState() {
		return open;
	}

	public void setState(boolean o) {
		open = o;
	}

	public void Wash(Apple apple) {
		if (open) {
			if (apple.getDirty()) {
				apple.setDirty(false);
			}
		}
	}
}
